docs/source/protocol_v3.rst

```diff
@@ -1,51 +1,437 @@
+
+.. code-block:: none
+
+   Header:
+      0 (int32) Disconnect
+      <val> (no type) Unused slot, any value acceptable
+
+
+Energies
+^^^^^^^^
+
+Sent from the simulation engine to the receiver each IMD frame if 
+energies were previously specified for this session in `Session info`_.
+
+.. note:: 
+  While the integration step is included in this
+  packet, this is a result of inheriting the IMD energy block from IMDv2. It is reccomended
+  to make use of the 64-bit integer integration step value from the time packet 
+  in analysis code instead.
+
+.. code-block:: none
+
+   Header:
+      1 (int32) Energies
+      1 (int32) Number of IMD energy blocks being sent
+
+   Body:
+      <val> (int32) Current integration step of the simulation
+      <val> (float32) Absolute temperature
+      <val> (float32) Total energy
+      <val> (float32) Potential energy
+      <val> (float32) Van der Waals energy
+      <val> (float32) Coulomb interaction energy
+      <val> (float32) Bonds energy
+      <val> (float32) Angles energy
+      <val> (float32) Dihedrals energy
+      <val> (float32) Improper dihedrals energy
+
+
+Coordinates
+^^^^^^^^^^^
+
+Sent from the simulation engine to the receiver each IMD frame if 
+coordinates were previously specified for this session in `Session info`_.
+
+.. code-block:: none
+
+   Header:
+      2 (int32) Coordinates
+      <n_atoms> (int32) Number of atoms in the IMD system
+
+   Body:
+      <array> (float32[n_atoms * 3]) X, Y, and Z coordinates of each atom in the 
+                                     IMD system encoded in the order 
+                                     [X1, Y1, Z1, ..., Xn, Yn, Zn]
+
+Go
+^^
+
+Sent from the receiver to the simulation engine after the receiver receives 
+the `Handshake`_ and `Session info`_ packets. 
+
+If the simulation engine does not 
+receive this packet within 1 second of sending the handshake and session info 
+packets, it should assume the receiver is incompatible. Whether the simulation engine
+exits or accepts another connection after this is an implementation decision.
+
+.. code-block:: none
+
+   Header:
+      3 (int32) Go
+      <val> (no type) Unused slot, any value acceptable
+
+
+Handshake
+^^^^^^^^^
+
+Sent from the simulation engine to the receiver after a socket connection
+is established. Unlike other header packets, the last four bytes of this packet are provided in 
+the native endianness of the sending simulation engine's hardware.
+
+The receiver can use this packet to determine both the IMD version
+of the session and the endianness of the simulation engine. By providing 
+the endianness of the machine running the simulation engine, the bulk of the 
+data being sent in the session, i.e. the body packets, do not have to be swapped 
+by the simulation engine before being sent, speeding up execution.
+
+.. code-block:: none
+
+   Header:
+      4 (int32) Handshake
+      3 (int32, unswapped byte order) IMD version used in session
```

Comment:
We need to say somewhere (probably at the top) that und Version numbers are monotonously increasing integers